muikku-timed-notifications-plugin/src/main/java/fi/otavanopisto/muikku/plugins/timed/notifications/strategies/StudyTimeNotificationStrategy.java

```diff
@@ -102,10 +103,19 @@ public void sendNotifications() {
       if (studentEntity != null) {
         User student = userController.findUserByIdentifier(studentIdentifier);
         
+        // Do not notify students that have no study start date set or have started their studies within the last 60 days
+        
         if (student.getStudyStartDate() == null || student.getStudyStartDate().isAfter(sendNotificationIfStudentStartedBefore)) {
           continue;
         }
         
+        // Double check that the study time really is ending in 60 days
+        
+        if (student.getStudyTimeEnd() == null || student.getStudyTimeEnd().isAfter(studyTimeEndsOdt) || student.getStudyTimeEnd().isBefore(OffsetDateTime.now())) {
```

Comment:
I guess all returned students have studytimeend specified as this would scream a bunch if it wasn't.